### ISSUE TITLE
[MRESOLVER-220] Modify signaling for unsupported operations

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resolver#946]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** opened **[MRESOLVER-220](https://issues.apache.org/jira/browse/MRESOLVER-220?redirect=false)** and commented

Lock upgrading is not supported, it is the caller that must be aware of this.

Currently the code for all implementations returns `false` in case of lock upgrade attempted, same way as locking failed. IMHO, we should change this to some exception maybe?

In short: code attempting unsupported "lock upgrade" should fail fast and with a clear reason ("illegal op"). Fail fast will happen with a runtime exception thrown and bubbling up all to the caller to make it clear, that there is some problem with code, that cannot be fixed without code change.

---

**Remote Links:**
- [GitHub Pull Request #301
](https://github.com/apache/maven-resolver/pull/301)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

An exception is a contract change and can only be done in version 2.0.0.
